packages/next/src/lib/flat-readdir.ts

```diff
@@ -1,29 +1,20 @@
 import { join } from 'path'
-import { nonNullable } from './non-nullable'
-import { promises } from 'fs'
+import fs from 'fs/promises'
 
 export async function flatReaddir(dir: string, includes: RegExp[]) {
-  const dirents = await promises.readdir(dir, { withFileTypes: true })
-  const result = await Promise.all(
-    dirents.map(async (part) => {
-      const absolutePath = join(dir, part.name)
-      if (part.isSymbolicLink()) {
-        const stats = await promises.stat(absolutePath)
-        if (stats.isDirectory()) {
-          return null
-        }
-      }
+  const dirents = await fs.opendir(dir)
+  const result = []
 
-      if (
-        part.isDirectory() ||
-        !includes.some((include) => include.test(part.name))
-      ) {
-        return null
-      }
+  for await (const part of dirents) {
+    const shouldOmit =
+      part.isDirectory() ||
+      part.isSymbolicLink() ||
```

Comment:
Does this handle the case when the symlink is found but its linking to a directory?

It seems like the previous code was explicitly calling `await promises.stat(absolutePath)` to handle this case.

Reply:
Yes, you're right. I've updated the pull request. Thank you for the review.